Mathlib/Algebra/Category/Grp/ChosenFiniteProducts.lean

```diff
@@ -0,0 +1,121 @@
+
+@[simp]
+theorem μ_forget_apply {G H : Grp.{u}} (p : G) (q : H) :
+    Functor.LaxMonoidal.μ (forget Grp.{u}) G H (p, q) = (p, q) := by
+  apply Prod.ext
+  · exact congrFun (Functor.Monoidal.μ_fst (forget Grp.{u}) G H) (p, q)
+  · exact congrFun (Functor.Monoidal.μ_snd (forget Grp.{u}) G H) (p, q)
+
+end Grp
+
+namespace AddGrp
+
+/-- Construct limit data for a binary product in `AddGrp`, using `AddGrp.of (G × H)` -/
+@[simps! cone_pt isLimit_lift]
+def binaryProductLimitCone (G H : AddGrp.{u}) : LimitCone (pair G H) where
+  cone := BinaryFan.mk (ofHom (AddMonoidHom.fst G H)) (ofHom (AddMonoidHom.snd G H))
+  isLimit := BinaryFan.IsLimit.mk _ (fun l r => ofHom (AddMonoidHom.prod l.hom r.hom))
+    (fun _ _ => rfl) (fun _ _ => rfl) (by aesop_cat)
+
+/-- We choose `AddGrp.of (G × H)` as the product of `G` and `H` and `AddGrp.of PUnit` as
+the terminal object. -/
+noncomputable instance : ChosenFiniteProducts AddGrp.{u} where
+  product G H := binaryProductLimitCone G H
+  terminal := ⟨_, (isZero_of_subsingleton (AddGrp.of PUnit.{u + 1})).isTerminal⟩
+
+attribute [local instance] Functor.monoidalOfChosenFiniteProducts
+
+theorem tensorObj_eq (G H : AddGrp.{u}) : (G ⊗ H) = of (G × H) := rfl
+
+@[simp]
+theorem μ_forget_apply {G H : AddGrp.{u}} (p : G) (q : H) :
+    Functor.LaxMonoidal.μ (forget AddGrp.{u}) G H (p, q) = (p, q) := by
+  apply Prod.ext
+  · exact congrFun (Functor.Monoidal.μ_fst (forget AddGrp.{u}) G H) (p, q)
+  · exact congrFun (Functor.Monoidal.μ_snd (forget AddGrp.{u}) G H) (p, q)
+
+end AddGrp
+
+namespace CommGrp
+
+/-- Construct limit data for a binary product in `CommGrp`, using `CommGrp.of (G × H)` -/
+@[simps! cone_pt isLimit_lift]
+def binaryProductLimitCone (G H : CommGrp.{u}) : LimitCone (pair G H) where
+  cone := BinaryFan.mk (ofHom (MonoidHom.fst G H)) (ofHom (MonoidHom.snd G H))
+  isLimit := BinaryFan.IsLimit.mk _ (fun l r => ofHom (MonoidHom.prod l.hom r.hom))
+    (fun _ _ => rfl) (fun _ _ => rfl) (by aesop_cat)
+
+/-- We choose `CommGrp.of (G × H)` as the product of `G` and `H` and `CommGrp.of PUnit` as
+the terminal object. -/
+noncomputable instance : ChosenFiniteProducts CommGrp.{u} where
+  product G H := binaryProductLimitCone G H
+  terminal := ⟨_, (isZero_of_subsingleton (CommGrp.of PUnit.{u + 1})).isTerminal⟩
+
+attribute [local instance] Functor.monoidalOfChosenFiniteProducts
+
+theorem tensorObj_eq (G H : CommGrp.{u}) : (G ⊗ H) = of (G × H) := rfl
+
+@[simp]
+theorem μ_forget_apply {G H : CommGrp.{u}} (p : G) (q : H) :
+    Functor.LaxMonoidal.μ (forget CommGrp.{u}) G H (p, q) = (p, q) := by
+  apply Prod.ext
+  · exact congrFun (Functor.Monoidal.μ_fst (forget CommGrp.{u}) G H) (p, q)
+  · exact congrFun (Functor.Monoidal.μ_snd (forget CommGrp.{u}) G H) (p, q)
+
+end CommGrp
+
+namespace AddCommGrp
+
+/-- We choose `AddCommGrp.of (G × H)` as the product of `G` and `H` and `AddGrp.of PUnit` as
+the terminal object. -/
+noncomputable instance : ChosenFiniteProducts AddCommGrp.{u} where
```

Comment:
This one could be a definition/local instance instance of an instance, because there is another reasonable monoidal structure on `AddCommGrp` given by the tensor product.

Reply:
Fixed. I hope that one day we will instead downgrade the instance `ChosenFiniteProducts -> MonoidalCategory` to a `def`.